### PR TITLE
Windows

### PR DESCRIPTION
**Pull request by loganhz**
Proposed changes
======
- Backend added a new field `windowsPreferedCluster` to indicate Windows cluster.
So we don't have to check the flannel params to see if the cluster is for Windows or not. And in current master build, the override toggle for Windows supports and Flannel backend mode is broken. And it will be very hard to fix it without backend changes. So in this PR, we will not allow users to override the Windows Supports and Flannel backend fields as it doesn't work at all. We will address it in issue https://github.com/rancher/rancher/issues/21910
- Hide Istio menus in Windows cluster due to Istio doesn't support Windows
- Show warning in catalog item list page for Windows clusters as not all apps can work in the Windows clusters.
- Update the warning message for host port limitation in Windows cluster
- Add warning message for Rancher logging
- Fix the issue that edit yaml cause Windows failed
- Update the default image for Windows cluster

Types of changes
======
- Bugfix (non-breaking change which fixes an issue)

Linked Issues
======
https://github.com/rancher/rancher/issues/21909
https://github.com/rancher/rancher/issues/20951
https://github.com/rancher/rancher/issues/20089
https://github.com/rancher/rancher/issues/20940
 

Further comments
======

### CODE REVIEW

File: app/components/container/form-ports/component.js [hunk 1]
```diff
@@ -196,6 +195,11 @@ export default Component.extend({
           translationKey: `formPorts.kind.LoadBalancerDisbaled`,
           disabled:       true
         })
+      } if (get(this, 'scope.currentCluster.isWindows') && k === 'HostPort') {
```
Comment (westlywright) [inline app/components/container/form-ports/component.js]: Can you fix the weird line here?

**Comment by westlywright** [comment]
With this PR removing support for windows override will your new issue and fixes also address https://github.com/rancher/rancher/issues/21557 ?

File: app/components/container/form-ports/component.js (hunk 1 above)
Comment (loganhz) [reply app/components/container/form-ports/component.js]: Done

**Comment by loganhz** [comment]
> With this PR removing support for windows override will your new issue and fixes also address [rancher/rancher#21557](https://github.com/rancher/rancher/issues/21557) ?

Yes, assigned to myself.

**Comment by loganhz** [comment]
@westlywright updated. Thanks

**Pull request merged**